uploadToFlow.sh

```diff
@@ -0,0 +1,51 @@
+#! /bin/bash
+
+die() {
+    >&2 echo "$*"
+    exit 1
+}
+
+require() {
+    command -v "$1" > /dev/null || die "$1 is not in the \$PATH"
+}
+
+if [ "$#" -ne "2" ] ; then
+    die "usage: $(basename "$0") SERVER ENTITY_FILE"
+fi
+
+server=$1
+entityFile=$2
+
+# Make sure that scripts we call are in the PATH.
+require uploadFlow.sh
+require uploadResourceCollection.sh
+require uploadTrigger.sh
+require uploadSharedConfig.sh
+require uploadRecipe.sh
+
+if [ ! -r "${entityFile}" ] ; then
+    die "can't read file: ${entityFile}"
+fi
+
+if [[ "${entityFile}" =~ flows/ ]] ; then
+    cd "$(dirname "${entityFile}")/../../.." || exit 1
+    uploadFlow.sh "$(basename "${entityFile%%.json}")" "${server}"
+
+elif [[ "${entityFile}" =~ flowResources/ ]] ; then
+    cd "$(dirname "${entityFile}")" || exit 1
+    uploadResourceCollection.sh "$(basename "${entityFile}")" "${server}"
+
+elif [[ "${entityFile}" =~ triggerers/ ]] ; then
+    cd "$(dirname "${entityFile}")/../../.." || exit 1
+    uploadTrigger.sh "$(basename "${entityFile%%.json}")" "${server}"
+
+elif [[ "${entityFile}" =~ sharedConfig/ ]] ; then
+    cd "$(dirname "${entityFile}")/../../.." || exit 1
+    uploadSharedConfig.sh "$(basename "${entityFile%%.json}")" "${server}"
+
+elif [[ "$(pwd)/${entityFile}" =~ recipe ]] ; then
```

Comment:
I read your explanation but I'm still a bit confused on the intention, is it in case your recipe is not in `ihub-recipes/recipes/...` and elsewhere (since you'd be providing absolute path)?  Because `recipe` would obviously match the ihub-recipes repository, but `recipes/` would be a little safer.  I understand it would not match different paths for recipes, but then you'd need to presently be in a directory that has the word 'recipe' in it anyway.

Reply:
I changed recipe to recipes/.  That sounds like it will make the script safer to run.  Thanks @devaidendale!